kernel/traffic/core/src/main/java/org/apache/shardingsphere/traffic/algorithm/traffic/segment/SQLRegexTrafficAlgorithm.java

```diff
@@ -41,8 +40,8 @@ public void init(final Properties props) {
         ShardingSpherePreconditions.checkState(props.containsKey(REGEX_PROPS_KEY),
                 () -> new SegmentTrafficAlgorithmInitializationException(SQLRegexTrafficAlgorithm.class.getName(), String.format("%s cannot be null", REGEX_PROPS_KEY)));
         regex = Pattern.compile(props.getProperty(REGEX_PROPS_KEY));
-        ShardingSpherePreconditions.checkState(Strings.isNullOrEmpty(String.valueOf(regex)),
-                () -> new SegmentTrafficAlgorithmInitializationException(SQLRegexTrafficAlgorithm.class.getName(), "regex must be not null or empty"));
+        ShardingSpherePreconditions.checkState(!Strings.isNullOrEmpty(String.valueOf(regex)),
```

Comment:
Hi @Netter99 @Pace2Car 
I'm wondering if the result of `props.getProperty(REGEX_PROPS_KEY)` is null, will `Pattern.compile` throw an exception?